### ISSUE TITLE
"conda install" ignores --no-update-dependencies

### ISSUE THREAD

**Issue by janssen** [conda/conda#2207]
```
% conda info
Current conda install:

             platform : osx-64
        conda version : 3.18.8
  conda-build version : not installed
       python version : 2.7.11.final.0
     requests version : 2.8.1
     root environment : /tilde/janssen/miniconda2  (writable)
  default environment : /tilde/janssen/miniconda2
     envs directories : /tilde/janssen/miniconda2/envs
        package cache : /tilde/janssen/miniconda2/pkgs
         channel URLs : https://repo.continuum.io/pkgs/free/osx-64/
                        https://repo.continuum.io/pkgs/free/noarch/
                        https://repo.continuum.io/pkgs/pro/osx-64/
                        https://repo.continuum.io/pkgs/pro/noarch/
          config file : None
    is foreign system : False

% conda install --no-update-dependencies --dry-run conda-build
Fetching package metadata: ....
Solving package specifications: .
Package plan for installation in environment /tilde/janssen/miniconda2:

The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    openssl-1.0.2g             |                0         3.0 MB  defaults
    sqlite-3.9.2               |                0         1.2 MB  defaults
    markupsafe-0.23            |           py27_0          22 KB  defaults
    requests-2.9.1             |           py27_0         604 KB  defaults
    setuptools-20.2.2          |           py27_0         453 KB  defaults
    wheel-0.29.0               |           py27_0          81 KB  defaults
    conda-4.0.1                |           py27_0         184 KB  defaults
    jinja2-2.8                 |           py27_0         263 KB  defaults
    pip-8.1.0                  |           py27_0         1.5 MB  defaults
    conda-build-1.19.0         |           py27_0         211 KB  defaults
    ------------------------------------------------------------
                                           Total:         7.5 MB

The following NEW packages will be INSTALLED:

    conda-build: 1.19.0-py27_0 defaults
    jinja2:      2.8-py27_0    defaults
    markupsafe:  0.23-py27_0   defaults
    wheel:       0.29.0-py27_0 defaults

The following packages will be UPDATED:

    conda:       3.18.8-py27_0 defaults --> 4.0.1-py27_0  defaults
    openssl:     1.0.2d-0      defaults --> 1.0.2g-0      defaults
    pip:         7.1.2-py27_0  defaults --> 8.1.0-py27_0  defaults
    requests:    2.8.1-py27_0  defaults --> 2.9.1-py27_0  defaults
    setuptools:  18.5-py27_0   defaults --> 20.2.2-py27_0 defaults
    sqlite:      3.8.4.1-1     defaults --> 3.9.2-0       defaults

Dry run: exiting
%
```

**Comment by mcg1969** [comment]
I agree that it should not be doing this. Can you do me a favor and copy and paste the full output of `conda --debug install --no-update-dependencies --dry-run conda-build`? (You might have to put the `--debug` at the end of the command)

**Comment by janssen** [comment]
Not sure you need me for this.  Just install conda (I used Miniconda2-latest), and then conda-build.

**Comment by janssen** [comment]
Here it is.  Again, this is Miniconda2 (latest) on OS X.
```
% conda --debug install --no-update-dependencies --dry-run conda-build
DEBUG:conda.config:Could not import binstar
DEBUG:conda.fetch:channel_urls=('https://repo.continuum.io/pkgs/free/osx-64/', 'https://repo.continuum.io/pkgs/free/noarch/', 'https://repo.continuum.io/pkgs/pro/osx-64/', 'https://repo.continuum.io/pkgs/pro/noarch/')
Fetching package metadata: INFO:stdoutlog:Fetching package metadata: 
DEBUG:requests.packages.urllib3.util.retry:Converted retries value: 3 -> Retry(total=3, connect=None, read=None, redirect=None)
INFO:requests.packages.urllib3.connectionpool:Starting new HTTPS connection (1): repo.continuum.io
DEBUG:requests.packages.urllib3.connectionpool:"GET /pkgs/pro/noarch/repodata.json.bz2 HTTP/1.1" 200 70
.DEBUG:dotupdate:fetching repodata: args ('https://repo.continuum.io/pkgs/pro/noarch/',) kwargs {'session': <conda.connection.CondaSession object at 0x10042cf50>, 'use_cache': False}
DEBUG:requests.packages.urllib3.connectionpool:"GET /pkgs/pro/osx-64/repodata.json.bz2 HTTP/1.1" 200 21096
.DEBUG:dotupdate:fetching repodata: args ('https://repo.continuum.io/pkgs/pro/osx-64/',) kwargs {'session': <conda.connection.CondaSession object at 0x10042cf50>, 'use_cache': False}
DEBUG:requests.packages.urllib3.connectionpool:"GET /pkgs/free/noarch/repodata.json.bz2 HTTP/1.1" 200 464
.DEBUG:dotupdate:fetching repodata: args ('https://repo.continuum.io/pkgs/free/noarch/',) kwargs {'session': <conda.connection.CondaSession object at 0x10042cf50>, 'use_cache': False}
DEBUG:requests.packages.urllib3.connectionpool:"GET /pkgs/free/osx-64/repodata.json.bz2 HTTP/1.1" 200 245985
.DEBUG:dotupdate:fetching repodata: args ('https://repo.continuum.io/pkgs/free/osx-64/',) kwargs {'session': <conda.connection.CondaSession object at 0x10042cf50>, 'use_cache': False}

INFO:stdoutlog:

DEBUG:conda.plan:Pinned specs=[]
DEBUG:conda.plan:H0 specs=['conda-build', 'conda']
DEBUG:conda.plan:H2 python True
DEBUG:conda.plan:H3 python
DEBUG:conda.plan:HF specs=['conda-build', 'conda', 'python 2.7*']
DEBUG:conda.resolve:specs=['conda-build', 'conda', 'python 2.7*']  features=set([])
Solving package specifications: INFO:stdoutlog:Solving package specifications: 
DEBUG:conda.resolve:Solving for ['conda-build', 'conda', 'python 2.7*']
DEBUG:conda.resolve:Features: set([])
DEBUG:conda.resolve:Installed: ['sqlite-3.8.4.1-1.tar.bz2', 'pyyaml-3.11-py27_1.tar.bz2', 'pip-7.1.2-py27_0.tar.bz2', 'tk-8.5.18-0.tar.bz2', 'zlib-1.2.8-0.tar.bz2', 'setuptools-18.8.1-py27_0.tar.bz2', 'yaml-0.1.6-0.tar.bz2', 'conda-env-2.4.5-py27_0.tar.bz2', 'pycosat-0.6.1-py27_0.tar.bz2', 'readline-6.2-2.tar.bz2', 'wheel-0.26.0-py27_1.tar.bz2', 'pycrypto-2.6.1-py27_0.tar.bz2', 'openssl-1.0.2d-0.tar.bz2', 'python-2.7.11-0.tar.bz2', 'requests-2.9.0-py27_0.tar.bz2', 'conda-3.19.0-py27_0.tar.bz2']
.DEBUG:dotupdate:Solving using max dists only
DEBUG:conda.logic:min_sat using alg: iterate
DEBUG:conda.logic:Iterate ran 2 times

Package plan for installation in environment /tmp/foobar:

The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    openssl-1.0.2g             |                0         3.0 MB
    sqlite-3.9.2               |                0         1.2 MB
    markupsafe-0.23            |           py27_0          22 KB
    requests-2.9.1             |           py27_0         604 KB
    setuptools-20.2.2          |           py27_0         453 KB
    wheel-0.29.0               |           py27_0          81 KB
    conda-4.0.2                |           py27_0         184 KB
    jinja2-2.8                 |           py27_0         263 KB
    pip-8.1.0                  |           py27_0         1.5 MB
    conda-build-1.19.0         |           py27_0         211 KB
    ------------------------------------------------------------
                                           Total:         7.5 MB

The following NEW packages will be INSTALLED:

    conda-build: 1.19.0-py27_0
    jinja2:      2.8-py27_0   
    markupsafe:  0.23-py27_0  

The following packages will be UPDATED:

    conda:       3.19.0-py27_0 --> 4.0.2-py27_0 
    openssl:     1.0.2d-0      --> 1.0.2g-0     
    pip:         7.1.2-py27_0  --> 8.1.0-py27_0 
    requests:    2.9.0-py27_0  --> 2.9.1-py27_0 
    setuptools:  18.8.1-py27_0 --> 20.2.2-py27_0
    sqlite:      3.8.4.1-1     --> 3.9.2-0      
    wheel:       0.26.0-py27_1 --> 0.29.0-py27_0

Dry run: exiting
%

**Comment by kalefranz** [comment]
miniconda2 latest ships with `3.19.0` right now I think.  It's also showing that in your output

    conda:       3.19.0-py27_0 --> 4.0.2-py27_0 

After installing miniconda, run `conda update conda` and then `conda --version` to ensure you're on conda `>= 4.0.2`.  Then try your command again.

**Comment by janssen** [comment]
So, I'd be testing your fix?  The problem is that the path you suggest (doing "conda update conda") will in fact update those packages (openssl, wheel, etc.) so that the bug doesn't exhibit itself, if it still exists.  It won't prove anything.  We'd need to find something other than conda-build to test with, something that requires updates.

**Comment by mcg1969** [comment]
Use the `--force` option, perhaps; i.e. `conda install --force conda=4.0.2`?

**Comment by kalefranz** [comment]
@janssen (and @mcg1969), I really don't understand this quite yet.  You're using conda `< 4.0` as your base conda, correct?  So you're testing the `--no-update-dependencies` flag in a version of conda that's less than 4.0... I think...

Are you asking for a fix to conda 3.19.x?

**Comment by janssen** [comment]
@kalefranz -- well, it would be neat if you could fix 3.19.x with a time machine :-), but no.  I was just pointing out that my running the tests wouldn't tell if you had fixed this bug.  This isn't blocking anything for me (except a workaround for the conda-build bug), so I think I can drop out of this issue and you folks can fix the bug (and test the fix) at your own schedule.

**Comment by kalefranz** [comment]
OK cool.  Working on getting the conda-build recursive fix out today.
